src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -84,6 +85,7 @@ public static function getSubscribedServices()
             'security.token_storage' => '?'.TokenStorageInterface::class,
             'security.csrf.token_manager' => '?'.CsrfTokenManagerInterface::class,
             'parameter_bag' => '?'.ContainerInterface::class,
+            'message_bus' => '?'.MessageBusInterface::class,
```

Comment:
So you can do a `$this->get('message_bus')` within your controllers.

Reply:
I don't think that we do that for any other service, nor that we want to promote that (we prefer to promote regular injection since 4.0). Other services are injected here mostly for backward compatibility.

Reply:
However, maybe can you add a proxy method in the trait, for convenience and discoverability by newcomers.

Reply:
Added the `dispatch` method on the `ControllerTrait` in 0a310798ee.